### ISSUE TITLE
Auto-regenerate AWS CodeArtifact temporary tokens

### ISSUE THREAD

**Issue by lnxslck** [dependabot/dependabot-core#5686]
### Package ecosystem

nodejs

### Language version

nodejs 14.1

### What you expected to see, versus what you actually saw

I expected dependabot to be able to generate CODEARTIFACT token automatically so it can be used for AWS Codeartifact private registry.

Instead, I need to generate the CODEARTIFACT_TOKEN from the CLI as I need to run aws codeartifact get-authorization-token.
This means that I need to generate the token, add it as a secret manually, the next day I need to do it all over again.
If dependabot.yaml worked like a Github Action we could use the run command.

**Comment by AndreiPaulau** [comment]
Yeah, would be great to extend dependabot functionality by adding AWS Secret / Access keys  secrets parameters for further CODEARTIFACT_TOKEN token generation.

**Comment by lnxslck** [comment]
Many companies generate these values, AWS keys and CODEARTIFACT per request and are temporary for security reasons.
It would be great if you could generate the CODEARTIFACT token inside dependabot just like a Github Actions run command.
That way, each time dependabot runs, generates a new CODEARTIFACT token and uses it.
This increases security tremendously instead of using the same tokens over and over again.

**Comment by jgtvares** [comment]
Any updates on this?
I'm enabling Dependabot in a bunch of repositories and all of them use private packages published in CodeArtifact.
Since the CA token expires in 12 hours, it's impracticable to change the token on every single repository.
It would be really nice to be able to run a command just before the Dependabot analysis like @lnxslck pointed out.

**Comment by jeffwidman** [comment]
A couple of thoughts:

For the Dependabot Service that GitHub runs natively, there's two options:
  1. Today the Dependabot secrets can be [rotated via an API](https://docs.github.com/en/rest/dependabot/secrets?apiVersion=2022-11-28). So you could script the rotation on your desired frequency... for example, if Dependabot runs weekly, you can script the token to rotate the day before with a 24 hour expiration, so it expires right after Dependabot runs.
  2. Long term, if CodeArtifact supports OIDC that would allow a more automated connection/rotation... we don't support that today, but may eventually... tracked in: https://github.com/dependabot/dependabot-core/issues/6580

Another option is that `dependabot-core` (this repo) is a library... so you can run it yourself using an entrypoint like https://github.com/dependabot/dependabot-script, perhaps within the context of a GitHub action. That would let you run another action to fetch a temporary token.

For passing that token to the registry client in `dependabot-core`, I haven't checked source, but I suspect it may be missing the plumbing to pass creds as arguments through the script and into core... however, a simpler pattern is to integrate with the AWS CodeArtifact SDK Ruby library and let it lookup well-known local env vars--examples of how we do it elsewhere here in `dependabot-core`:
  * https://github.com/dependabot/dependabot-core/pull/1557
  * https://github.com/dependabot/dependabot-core/pull/1463
  * We'd happily accept a community PR here to add something like ☝️ for CodeArtifact.
